src/historical.rs

```diff
@@ -202,6 +202,136 @@ impl From<(time::OffsetDateTime, time::Duration)> for DateTimeRange {
     }
 }
 
+#[cfg(feature = "chrono")]
+fn chrono_datetime_to_time(dt: chrono::DateTime<chrono::Utc>) -> time::OffsetDateTime {
+    // timestamp_nanos_opt() returns None for dates outside ~1677-2262.
+    // from_unix_timestamp_nanos() fails for dates outside ~1000-9999.
+    // Practical timestamps fall well within these bounds, so unwrap is safe.
+    time::OffsetDateTime::from_unix_timestamp_nanos(dt.timestamp_nanos_opt().unwrap() as i128)
+        .unwrap()
+}
+
+#[cfg(feature = "chrono")]
+fn chrono_date_to_time(date: chrono::NaiveDate) -> time::Date {
+    use chrono::Datelike;
+    time::Date::from_calendar_date(
+        date.year(),
+        time::Month::try_from(date.month() as u8).unwrap(),
+        date.day() as u8,
+    )
+    .unwrap()
+}
+
+#[cfg(feature = "chrono")]
+impl<Tz: chrono::TimeZone> From<std::ops::Range<chrono::DateTime<Tz>>> for DateTimeRange {
```

Comment:
I'd move the all the impls into a `mod` block so you can have a single `cfg(feature...)` in this file like what you did with the tests